Staff

prphawk opened the pull request:
pf aprovem e nao liguem pro que ta inacabado pq eu preciso mergear esse negocio de uma vezzzzzz

prphawk committed:
Auto stash before merge of "EC_sync" and "origin/context_improvements"

prphawk committed:
Merge remote-tracking branch 'origin/context_improvements' into EC_sync

prphawk committed:
Merge remote-tracking branch 'origin/context_improvements' into EC_sync

prphawk committed:
Feat(User Permission)

prphawk committed:
Merge branch 'master' into login

prphawk committed:
Feat(User Permission Enum)

prphawk committed:
Feat(Private Route based on User)

prphawk committed:
Feat(Centralizing Redirect Service)

canssadíssima de debugar viu tudo teria sido eitado se eu n fosse uma idiota

prphawk committed:
remove(unused vars)

prphawk committed:
Feat(+ Staff Components)

prphawk committed:
Feat(Staff Moderation Component)

prphawk committed:
Feat(TabPanel)

prphawk committed:
Feat(Staff Moderation Panels)

prphawk committed:
Remove(swipe package)

prphawk committed:
Feat(Staff Everything and Reactive Component List)

prphawk committed:
Feat(styles and views)

prphawk committed:
Fix(Render Form, User API Bug)

prphawk committed:
Feat(FormContent Component Modularized)

prphawk committed:
Remove(default booleans)

prphawk committed:
Feat(Gambiarra nos controles do form)

prphawk committed:
Feat(Styles components str languages etc)

prphawk committed:
Feat(Refactor Contacts Component to handle Staff Presence)

prphawk committed:
Feat(Add/Edit/Delete EC as staff)

prphawk committed:
Feat(isStaff on Context)

prphawk committed:
Rename(React Component IsStaff)

prphawk committed:
Refactor(Contacts and ItemListMenu Components)

prphawk committed:
Refactor(Class Names)

prphawk committed:
Feat(Suport for CLIENT type of user): and relocating path functions

prphawk committed:
Feat(Treatment Component for STAFF management)

prphawk committed:
Feat(uuuh More Components Stuff)

prphawk committed:
Feat(Dialog Refactor)

prphawk committed:
Feat(Config Refactor)

prphawk committed:
Feat(Notes Dialog Refactor)

prphawk committed:
Feat(Dialog Refactor Complete)

prphawk committed:
Delete(unused vars)

prphawk committed:
Feat(STAFF Add GlossaryItem Button + Form)

prphawk committed:
Feat(Refactor Add Button Component)

prphawk committed:
Feat(just correcting names)

prphawk committed:
Feat(Glossary ListItem + Edit/Delete as STAFF)

prphawk committed:
Feat(uuuh the staff part i forgot)

prphawk committed:
Feat(Staff Mod Functions by ADMIN pt.1)

prphawk committed:
Feat(Staff Mod Functions by ADMIN pt. 2): edit/delete

prphawk committed:
Feat(component changes)

prphawk committed:
Feat(fewer useStates, wohoo!)

prphawk committed:
Remove(Warnings)

prphawk committed:
Feat(STAFF FAQ pt 1))

prphawk committed:
Delete(FAQ): a entidade não tem uso se Treatment existe

prphawk committed:
Fix(warnings)

prphawk committed:
Fix(classes)

prphawk committed:
Feat(FaqItems Router Params)

prphawk committed:
Feat(User & Staff Faq Route)

prphawk committed:
Feat(Blocking non-STAFF from faq/${localId} URL): description

only STAFF will be able to see all the F.A.Qs on the App

prphawk committed:
Feat(Treatment Functions and Optimizations :])

prphawk committed:
Feat(FAQ STAFF Functions)

JPedroSilveira committed:
Merge pull request #108 from petcomputacaoufrgs/delete_FAQ

Delete FAQ

prphawk committed:
Feat(Organizing)

prphawk committed:
Feat(Staff card & searchBar)

prphawk committed:
Merge remote-tracking branch 'origin/master' into 🤠staff_components

prphawk committed:
Fix(Merge)

prphawk committed:
Merge branch 'master' into 🤠staff_components

prphawk committed:
Fix(Merge)

prphawk committed:
Merge remote-tracking branch 'origin/master' into 🤠staff_components

prphawk committed:
Fix(Merge)

prphawk committed:
Feat(Translations pt.1)

prphawk committed:
Feat(Translations pt. 2)

prphawk committed:
Feat(MAX Length Form Validations pt. 1)

prphawk committed:
Refactoring(App Data Constants)

prphawk committed:
Feat(MAX Length Form Validations pt. 2)

prphawk committed:
Fix(design on accordion components): faq & glossary

prphawk committed:
Feat(MAX Length Form Validations pt. 3)

prphawk committed:
Feat(Main Component Refactor)

prphawk committed:
just changing some .css don't mind me

prphawk committed:
Refactor(API Path Constants)

prphawk committed:
Feat(Treatment Question Component)

prphawk committed:
Feat(TreatmentQuestionView)

JPedroSilveira reviewed (changes requested):
Bugs relatados no trello

JPedroSilveira committed:
Feat(FontSize): Add

prphawk committed:
Feat(Staff Data + Treatment and Faq View)

prphawk committed:
Refactor(FAQ Hub)

prphawk committed:
Feat(Changing .css and fixing bugs)

prphawk committed:
Merge remote-tracking branch 'origin/🤠staff_components' into 🤠staff_components

prphawk committed:
Feat(Change .css classNames)

prphawk committed:
Feat(Unique Treatment)

prphawk committed:
Feat(styles alterations)

prphawk committed:
Feat(Invalidation of repeated values): on GlossaryItem and Email

prphawk committed:
Feat(Essential Contact List of Treatments): on its card

prphawk committed:
Feat(Fix treatments edit, refactoring contacts)

prphawk committed:
Feat(Design EContact Card)

JPedroSilveira commented on src/App.tsx:
```diff
@@ -40,13 +44,14 @@ const App: React.FC = () => {
 			}
-
 			await loadTabInfo()
+			updateAuth(isAuthenticated)
```
Função sendo chamada duas vezes, primeiro em loadAuth()

JPedroSilveira commented on src/App.tsx:
```diff
@@ -76,6 +81,11 @@ const App: React.FC = () => {
 		}
 
+		const loadUserPermission = async () => {
+			const hasUserPermission = await UserService.getPermission()
```
Faltou adicionar o listenner de UserService em: 
		"AuthService.addUpdateEventListenner(loadData)
		UserSettingsService.addUpdateEventListenner(loadSettings)
		TabControlService.addUpdateEventListenner(loadData)"

JPedroSilveira replied:
Talvez userPermission seria melhor, por ser um enumerados com diferentes tipos de usuário

JPedroSilveira commented on src/App.css:
```diff
@@ -19,9 +19,16 @@ body .MuiInputBase-root {
 body .MuiListItemText-root .MuiTypography-body1 {
 	font-size: var(--default-font);
-	white-space: nowrap;
+	/* white-space: nowrap; */
```
Remover trecho comentado

JPedroSilveira commented on src/components/dialogs/dino_dialog/props.ts:
```diff
@@ -0,0 +1,7 @@
+	onClose: () => void
+	onSave: () => void
+	header?: React.ReactNode
+	actions?: React.ReactNode
```
Tem que ver com o Afonso que tipo seria ideal aqui.
https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement
Tem essa discussão aqui sobre os tipos, acho que ReactNode é muito genérico, mas entre ReactElement e JSXElement eu não sei.

JPedroSilveira commented on src/components/settings/first_settings_dialog/index.tsx:
```diff
@@ -209,7 +188,7 @@ const FirstSettingsDialog: React.FC = () => {
 		if (settings && settings.language !== newLanguage) {
 			settings.language = newLanguage
-			UserSettingsService.saveOnlyLocally(settings)
+			UserSettingsService.save(settings)
```
Colocando save aqui ele já salva essa configuração na API. Se você removeu o save de cada step na API para evitar que isso ocorra então teria que manter o saveOnlyLocally aqui também e nos outros locais. (Nesse caso só salvaria o resultado final na API).

JPedroSilveira committed:
Feat(APIPaths): Reorganização das classes com os paths da API

JPedroSilveira commented on src/constants/api/APIRequestMappingConstants.ts:
```diff
@@ -33,7 +36,6 @@ class APIRequestMappingConstants {
-	TREATMENT_ESSENTIAL_CONTACTS = `${this.CONTACT}essential/faq/`
 	DELETE_ACCOUNT = `${this.USER}delete_account/`
 }
 
```
Abri uma PR para sua revisão nesta branch com uma proposta de renomeação das classes:
 - Uma para paths de HTTP
 - Uma para paths de WebSocket (fica aberta para mudanças que influenciem só o WebSocket)
 - Uma geral com os paths principais

JPedroSilveira commented on src/constants/app_data/DataConstants.ts:
```diff
@@ -0,0 +1,36 @@
+
+}
+
+export default DataConstants
```
Depois do merge vamos tentar padronizar assim no resto do App, achei lindo

JPedroSilveira commented on src/services/auth/AuthService.ts:
```diff
@@ -354,6 +358,17 @@ class AuthService extends UpdatableService {
 		await UserService.updateUser(responseBody.user)
 	}
+	
+	private async saveUser(responseBody: AuthResponseDataModel) {
```
Substitui a função "saveUserAuthData" por essa, fazem a mesma coisa

JPedroSilveira commented on src/types/contact/view/ContactView.ts:
```diff
@@ -1,4 +1,5 @@
 import ContactEntity from '../database/ContactEntity'
+import EssentialContactEntity from '../database/EssentialContactEntity'
```
Não usado

JPedroSilveira commented on src/views/main/contacts/contact_dialog_card/content/index.tsx:
```diff
@@ -89,8 +68,60 @@ const ContactCardContent = ({ item }: ContactCardContentProps) => {
+	return IsStaff() ? (
+		<List component='nav'>
+			<SectionTitle/>
+			{ isUniversal ? renderTreatmentListItem('Este contato é universal e aparecerá em todas as agendas de usuários')
```
Adicionar ao language

JPedroSilveira commented on src/views/main/faq_hub/question_dialog_form/styles.css:
```diff
@@ -1,7 +1,4 @@
+
 
 .dialog_form .MuiDialogContent-root .MuiDialogContent-dividers {
 	padding: 0.5rem 1.5rem !important;
```
Não referencia nenhum item

JPedroSilveira commented on src/views/main/glossary/styles.css:
```diff
@@ -3,3 +3,8 @@
 	height: calc(100% - 3.75rem);
 	width: 100%;
 }
+/* .glossary__card_title .icon_button__options {
```
Remover trecho não utilizado

JPedroSilveira commented on src/views/staff/treatment/treatment_form/index.tsx:
```diff
@@ -0,0 +1,85 @@
+            label={langData.STAFF_ADD_TREATMENT_NAME}
+            type='name'
+            value={item.name}
+            /* é importante descontruir o objeto ao atualizá-lo, mesmo com um único atributo, senão ele vira outro. */
```
Remover comentário

JPedroSilveira commented on src/views/staff/treatment/treatment_list_items/treatment_list_item/index.tsx:
```diff
@@ -0,0 +1,52 @@
+          badgeContent={questions.length}
+          variant="dot"
+        > 
+          {/* TODO: Como fazer pra mudar a aba a partir dum link??? */}
```
Como assim?

JPedroSilveira committed:
Fix(Note): Correções no CSS e melhoria na pesquisa de anotações

JPedroSilveira committed:
Fix(Quebra Texto): Correção em quebras nas labels do material UI

Separei o CSS do app em 3 arquivos.
General -> Coisas utilizadas em lugares espalhados do sistema
Material -> Coisas relativas a mudanças no material ui (fácil de saber onde encontrar)
App -> Coisas gerais do app (basicamente fonte e tela)

JPedroSilveira committed:
Fix(Serviço): Mudança no número de serviço

prphawk committed:
Merge pull request #126 from petcomputacaoufrgs/staff_components_paths

Feat(APIPaths): Reorganização das classes com os paths da API

JPedroSilveira committed:
Fix(GlossaryList): Adição de fontes padronizadas

JPedroSilveira committed:
Fix(Merge)

JPedroSilveira committed:
Fix(Menu): Tamanhos de fonte padronizados

JPedroSilveira committed:
Fix(Moderação & Tratamento): Tamanho das fontes

JPedroSilveira committed:
Feat(Grants): Adicionar validação de permissões sobre requisições

JPedroSilveira committed:
Feat(Authorities): Resumo para permitir todos os usuários e mudanças de rotas

JPedroSilveira committed:
Refactor(AuthEnum -> PermissionEnum): todas as variáveis citavam como permission

JPedroSilveira committed:
Fix(RemoveGoogleContact): Remoção de GoogleContact no frontend

JPedroSilveira committed:
Feat(Contatos/Telefones Essenciais): Finalização da sincronização de contatos e telefones essenciais

JPedroSilveira committed:
Feat(ContatosEssenciais): Criação se contatos conforme mudanças de tratamento

JPedroSilveira committed:
Feat(StaffQueue): Mudança de topic para queue no websocket devido a staff

JPedroSilveira committed:
Merge branch 'grants' into 🤠staff_components

JPedroSilveira committed:
Fix(FAQ): Atualização de tela ao selecionar um tratamento pela primeira vez

JPedroSilveira committed:
Fix(DeleteTreatment): Implementação de lógica para remoção de treatment

prphawk committed:
Fix(TreatmentView from localId bug)

prphawk committed:
Modularização de componentes FAQ

JPedroSilveira committed:
Bug(Admin/Staff): Bug de sincronização

Moderação -> Usuário convidado e cadastrado permanece como pendente para outra conta e sem etiqueta para si mesmo

prphawk committed:
Feat(Restrict Staff Moderation to ADMIN)

prphawk committed:
Merge remote-tracking branch 'origin/🤠staff_components' into 🤠staff_components

JPedroSilveira committed:
Bug(TreatmentQuestion): Perguntas somem no recarregar da página

JPedroSilveira committed:
Feat(Treatment Question): Staff answer

JPedroSilveira committed:
Feat(Treatment): Remove test stuff

JPedroSilveira committed:
Feat(FAQ): redirect to user questions tab

prphawk committed:
Merge branch 'main' into 🤠staff_components

prphawk committed:
Fix(Merge Errors): And oh boy, do we have them

JPedroSilveira committed:
Fix(Database): Atualização da versão do BD

JPedroSilveira committed:
Fix(UserSettings): rename of attribute settingsStep to step

prphawk committed:
Fix(FirstDialogs bug)

prphawk committed:
Feat(Test Instances)

prphawk committed:
Feat(FirstSetting Dialogs modularization): dialogo != lógica

prphawk committed:
Feat(check UserPermission to load test instances)

prphawk committed:
Feat(Getting rid of literals on FS_Dialog)

prphawk committed:
some more instances

prphawk committed:
Feat(+ test instances): so falta faqItem que por algum motivo não salva

prphawk committed:
Feat(hide setting components from staff)

prphawk committed:
Fix(Quando temos perguntas iguais, todas expandem e retraem juntas)

prphawk committed:
Feat(change FS dialogs by user permission)

prphawk committed:
Feat(corrigindo DataConstants de Password & Arrumando httpPaths)

prphawk committed:
stash

prphawk committed:
Feat(Design Fixes): nos dialogos de kids, cores, icones de contatos

tudo no trello

prphawk committed:
Feat(Less states with CRUD values)

prphawk committed:
Feat(tradução da faq que faltava)

prphawk committed:
Feat(No items on list UI indicator)

prphawk committed:
Feat(allowing treatmentQuestion edit and reusing faqItem form component)

prphawk committed:
designs designs designs

prphawk committed:
Fix(mais problemas de cores e design): pls release me from my suffering

prphawk committed:
Delete(a lot of unnecessary stuff from the treatmentQuestion view)

prphawk committed:
Feat(Filter List Component)

prphawk committed:
Feat(Filter Contacts List)

prphawk committed:
Feat(contact filter for staff + translations)

prphawk committed:
Feat(treatment questions filter + translations)

prphawk committed:
Feat(translations and removing literals)

prphawk committed:
Merge branch 'main' into 🤠staff_components

prphawk committed:
Feat(refactoring and contact card actions bug fixing)

prphawk committed:
Feat(refactoring and contact card actions bug fixing)

o problema era na ordem de execução das funções de onEdit e onDelete dentro do componente ItemListMenu

prphawk committed:
Merge remote-tracking branch 'origin/🤠staff_components' into 🤠staff_components

prphawk committed:
Feat(DinoSlider)

prphawk committed:
mais coisinhas do jogo

prphawk committed:
Feat(removendo os 0)

prphawk committed:
Feat(KidsSpace refactor and design)

prphawk committed:
Feat(DinoSlider game over function, restart and refactor)

prphawk committed:
center dialog and fix boolean

prphawk committed:
removing unnecessary stuff

prphawk committed:
fix(delete treatment not working)
